docs/contributing.rst

```diff
@@ -6,78 +6,157 @@ There are many ways to contribute.  Here are a few:
 #. Subscribe to `the mailing lists
    <https://www.open-mpi.org/community/lists/ompi.php>`_ and become
    active in the discussions.
-#. Obtain `a Git clone <https://www.open-mpi.org/source/>`_ of Open
+
+#. Obtain `a Git clone <https://github.com/open-mpi/ompi/>`_ of Open
    MPI's code base and start looking through the code.
 
    .. note:: Be sure to see the :doc:`Developers guide
-             </developers/index>` for technical details about the code
-             base and how to build it).
+             </developers/index>` for technical details about the
+             code base and how to build it).
 
-#. Write your own components and contribute them back to the main code
-   base.
-#. Contribute bug fixes and feature enhancements to the main code
+#. Report bug fixes to the main code base.
+
+   * First, ensure the bug was not already reported by searching
+     `existing      GitHub Issues
+     <https://github.com/open-mpi/ompi/issues>`_.
+   * If you're unable to find an open issue addressing the problem,
+     `open a new one <https://github.com/open-mpi/ompi/issues/new>`_.
+
+#. Submit bug fixes to the main code base.
+
+   * Awesome!  Open a new GitHub pull request with the patch.
+   * Ensure the PR description clearly describes the problem and
+     solution. If there is an existing GitHub issue open describing
+     this bug, please include it in the description so we can track
+     them together.
+   * Be sure to see the :ref:`Open source contributions
+     <contributing-open-source-label>` section, below.
+
+#. Submit feature enhancements and/or new components to the main code
    base.
+
+   * Awesome!  We love new ideas!
+   * You might want to suggest your change on the `devel mail list
+     <https://www.open-mpi.org/community/lists/ompi.php>`_ before you
+     start writing code.  The `developer level technical information
+     on the internals of Open MPI
+     <https://www.open-mpi.org/faq/?category=developers>`_ may also be
+     useful for large scale features.
+   * If you're contributing a large new piece of functionality, that
+     will be best viewed if you |mdash| or someone, anyone |mdash| is
```

Comment:
sorry - what is `|mdash|`?

Reply:
It's the macro we established in `conf.py` for a Unicode "em dash" -- i.e., a long dash.

Reply:
Ah, thanks.